src/vs/workbench/contrib/inlineChat/browser/inlineChatActions.ts

```diff
@@ -69,13 +69,21 @@ export class StartSessionAction extends Action2 {
 			},
 			icon: START_INLINE_CHAT,
 			menu: [{
-				id: MenuId.ChatTitleBarMenu,
-				group: 'a_open',
+				id: MenuId.EditorContext,
+				// Place below "Add Selection to Chat" (order 1) and "Add File to Chat" (order 2)
+				// which both use the '1_chat' group in the editor context menu
+				group: '1_chat',
 				order: 3,
+				when: ContextKeyExpr.and(
+					ContextKeyExpr.or(CTX_INLINE_CHAT_HAS_AGENT, CTX_INLINE_CHAT_HAS_AGENT2),
+					CTX_INLINE_CHAT_POSSIBLE,
+					EditorContextKeys.writable,
+					EditorContextKeys.editorSimpleInput.negate()
+				)
```

Comment:
Why the extra `when` condition that was not present before in `MenuId.ChatTextEditorMenu`?